### ISSUE TITLE
Terminal colors not reset

### ISSUE THREAD

**Issue by mixmastamyk** [jonathanslenders/python-prompt-toolkit#214]
Hi, awesome toolkit !!!!

Having a problem on Windows though, in just the hello world app + a while loop.  The prompt text gets set to bold.  After a few keystrokes everything gets set to bold.  Exiting does not clear the bold setting either.  So the whole command prompt window has had its settings changed permanently.

To get things back to normal, I need to run the "color" command in the command prompt afterwards.

Sounds like the "reset/normal style" command is getting lost somewhere.

Python 2.7.7
Windows 7

**Comment by jonathanslenders** [comment]
Hi @mixmastamyk,
This looks indeed like a bug. I'll try to investigate and fix before the next release.
Jonathan

**Comment by jonathanslenders** [comment]
Hi @mixmastamyk,
Could you send me some example code? I have trouble reproducing this issue.

**Comment by evanunderscore** [comment]
I've noticed that the examples (specifically, I ran autocompletion.py with no modification) leave my console text bright white instead of the usual grey. I'm using Windows 10, Python 2.7.11.

**Comment by mixmastamyk** [comment]
Hi, is the info from evanunderscore good enough?  I don't believe I did anything but run hello world under a while loop, though it has been a few days.

**Comment by mixmastamyk** [comment]
Here is a screenshot:
![screenshot](https://cloud.githubusercontent.com/assets/767039/12524749/2aab0d38-c114-11e5-9543-9f445261ba6a.png)

**Comment by jonathanslenders** [comment]
Thank you for the screenshots, that is very helpful!
Currently I'm still unable to reproduce it. My Windows machine is Windows 7. ``ver`` returns the same Version info as yours. But for some reason I don't get anything in bold.
I'll try to get a Windows 10 machine asap to do some testing there.

**Comment by jonathanslenders** [comment]
The cause could be that "15" is not the right default color. We call the following line when leaving the prompt:

https://github.com/jonathanslenders/python-prompt-toolkit/blob/master/prompt_toolkit/terminal/win32_output.py#L166

Would it work if that 15 is replaced by 7? (That is low intensity white.)

**Comment by evanunderscore** [comment]
That would solve the issue for me.

Alternatively you could save off the colors on startup and restore them to whatever they were on exit, but that's probably more trouble than it's worth.

**Comment by mixmastamyk** [comment]
Hmm, saving and restoring is probably the correct solution.  The reason is that some people change the color of their command-prompt permanently from grey on black, and if you hard-code the color then it will leave the console in an odd state at exit.

I think ansi codes have the concept of default color, but if win-api doesn't then the above sounds reasonable.

**Comment by mixmastamyk** [comment]
Forgot to mention that yes, grey is better than white in the common case, if you are pressed for time.

**Comment by nagya11** [comment]
Hmm, saving and restoring is probably the ???

**Comment by jonathanslenders** [comment]
Let's go for saving and restoring, that is clearly better. :-)

**Comment by evanunderscore** [comment]
It looks like colorama takes the save and restore approach: https://github.com/tartley/colorama/blob/master/colorama/winterm.py

**Comment by jonathanslenders** [comment]
Thanks for the link to colorama! I'm going to take that approach.

**Comment by jonathanslenders** [comment]
Hi everyone,

The following change should probably fix this issue: https://github.com/jonathanslenders/python-prompt-toolkit/commit/36114a366b8aedfc94cc5626830a0836b00d78b4

Could you please see whether it works now?
Jonathan

**Comment by mixmastamyk** [comment]
Yes, I'll try tomorrow--getting late tonight unfortunately.

**Comment by evanunderscore** [comment]
Working for me. Thanks!

**Comment by mixmastamyk** [comment]
Yes, fixes the main issue, leaving the console in a different state. Thank you.

There are some small nitpicks remaining. I'm not sure they matter much, but here goes.  On Linux the default prompt is bold (white), while the input text typed is normal (grey).  But on Windows, both the prompt and the text input are bold (white).

Also the fallback is still color 15, but I believe the command prompt comes installed by default using grey, not white.  Is this still the case?  (I've never used Win 8, 10.)  Perhaps you've changed your prompt to white and that is why white seems "normal".

**Comment by jonathanslenders** [comment]
Ok, thanks for the feedback! I'll keep this issue open, until those small bugs are fixed as well.

(For some reason, I have 15 as default. But I don't remember I ever changed anything.)
